packages/solo/src/pipe-entrypoint.js

```diff
@@ -0,0 +1,62 @@
+/* global process */
+// @ts-check
+import '@endo/init/pre-bundle-source.js';
+import 'node-lmdb';
+import '@endo/init';
+
+import { parse, stringify } from '@endo/marshal';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+import '@agoric/cosmic-swingset/src/anylogger-agoric.js';
+import { connectToFakeChain } from '@agoric/cosmic-swingset/src/sim-chain.js';
+
+// console.error('getting pipe entrypoing started');
+const [method, ...margs] = process.argv.slice(2);
+
+const { send: psend } = process;
+assert(psend);
+const send = (...args) => psend.apply(process, args);
+
+process.on('SIGINT', () => {
+  // console.log('exiting pipe child');
+  process.exit(99);
+});
+
+const main = async () => {
+  let baton = makePromiseKit();
```

Comment:
I regret not calling this pattern `mutex` in XSnap.

Reply:
I have the chance to, so now I did.